### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by jamesfredley**
Update dependencies to resolve build issue

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -57,6 +57,7 @@ dependencies {
     implementation "org.grails:grails-datastore-gorm:$datastoreVersion"
 
     compileOnly "jakarta.servlet:jakarta.servlet-api:$servletApiVersion" // Provided
+    compileOnly "jakarta.persistence:jakarta.persistence-api:$jakartaPersistenceVersion"
```
Comment (matrei) [inline build.gradle]: Why is this dependency added? I can't find any usage of it it in the project.

File: gradle.properties [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
-grailsGradlePluginVersion=6.2.0
-datastoreVersion=7.3.4
+grailsGradlePluginVersion=7.0.0-SNAPSHOT
+datastoreVersion=9.0.0-SNAPSHOT
 groovyVersion=4.0.22
 servletApiVersion=6.0.0
 spockVersion=2.3-groovy-4.0
 springVersion=6.1.12
 commonsLang3Version=3.14.0
+jakartaPersistenceVersion=3.1.0
```
Comment (matrei) [inline gradle.properties]: See question about this dependency in `build.gradle`

File: build.gradle (hunk 1 above)
Comment (jamesfredley) [reply build.gradle]: Without it the build was failing all of the sudden: https://github.com/grails/grails-plugin-converters/actions/runs/10818461245/job/30014131537

```
> Task :compileGroovy


FAILURE: Build failed with an exception.

> Task :compileGroovy FAILED
* What went wrong:
2 actionable tasks: 2 executed
Execution failed for task ':compileGroovy'.
> jakarta/persistence/Entity
```

Comment (matrei) [reply build.gradle]: Looking at the `compileClasspath` from that run (https://ge.grails.org/s/2q4bac7ymiggk/dependencies?toggled=W1swXSxbMCwyXSxbMCwyLFsyOTNdXV0) it seems `org.grails:grails-datastore-gorm:7.3.4` was used.

Comment (jamesfredley) [reply build.gradle]: Yes, that does match the version before this PR:  https://github.com/grails/grails-plugin-converters/pull/69/files#diff-3d103fc7c312a3e136f88e81cef592424b8af2464c468116545c4d22d6edcf19L4

Comment (matrei) [reply build.gradle]: Did you try to run the build without jakarta-persistance-api?

Comment (jamesfredley) [reply build.gradle]: I did locally and reproduced this issue and then resolved it initially with `compileOnly "jakarta.persistence:jakarta.persistence-api:$jakartaPersistenceVersion"`

In retesting through, I have found that `grailsGradlePluginVersion=7.0.0-SNAPSHOT` also resolved the issue, so I will remove the extra dependency which is not need with the new Gradle plugin version.

**Review by matrei — APPROVED** [review]

**Pull request merged**